Mathlib/Computability/EpsilonNFA.lean

```diff
@@ -269,10 +269,10 @@ theorem toNFA_evalFrom_match (start : Set σ) :
 theorem toNFA_correct : M.toNFA.accepts = M.accepts :=
   rfl
 
-theorem pumping_lemma [Fintype σ] {x : List α} (hx : x ∈ M.accepts)
-    (hlen : Fintype.card (Set σ) ≤ List.length x) :
+theorem pumping_lemma [Fintype σ] [DecidableEq σ] {x : List α} (hx : x ∈ M.accepts)
+    (hlen : Fintype.card σ ≤ List.length x) :
     ∃ a b c, x = a ++ b ++ c ∧
-      a.length + b.length ≤ Fintype.card (Set σ) ∧ b ≠ [] ∧ {a} * {b}∗ * {c} ≤ M.accepts :=
+      a.length + b.length ≤ Fintype.card σ ∧ b ≠ [] ∧ {a} * {b}∗ * {c} ≤ M.accepts :=
   M.toNFA.pumping_lemma hx hlen
```

Comment:
```suggestion
      a.length + b.length ≤ Fintype.card σ ∧ b ≠ [] ∧ {a} * {b}∗ * {c} ≤ M.accepts := by
  classical exact M.toNFA.pumping_lemma hx hlen
```

Reply:
Applied, should the `NFA.pumping_lemma` also hide its `DecidableEq` assumption like that?

Reply:
Yes, exactly!

Reply:
Okay, I've done that as well. I think this PR should be ready, could you take another look?